oauth/oauth_config.go

```diff
@@ -0,0 +1,119 @@
+//
+// Copyright (c) 2021 Red Hat, Inc.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package oauth
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/redhat-appstudio/service-provider-integration-operator/pkg/logs"
+	"github.com/redhat-appstudio/service-provider-integration-operator/pkg/spi-shared/oauthstate"
+	"golang.org/x/oauth2"
+	corev1 "k8s.io/api/core/v1"
+	kuberrors "k8s.io/apimachinery/pkg/api/errors"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+const (
+	oauthCfgSecretLabel             = "service-provider-integration/service-provider-config"
```

Comment:
We use a different naming convention for labels, see: https://github.com/redhat-appstudio/service-provider-integration-operator/blob/89f1d64fb36772c5694e0ff0645206befd847275/controllers/labels.go#L17

Also the expected value of this label is the type of the service provider, which to me is a little bit confusing given the name of the label.

What about calling this just simply `spi.appstudio.redhat.com/service-provider-type`?

Reply:
changed to `v1beta1.SPIAccessTokenLinkLabel`